### PR TITLE
monitoring(repo-updater): add duration on critical rate limit alerts, update next steps

### PR DESCRIPTION
**Pull request by bobheadxi**
For https://github.com/sourcegraph/sourcegraph/issues/36434#issuecomment-1148000738

1. @michaellzc noted that the pod restart advice is not great: 
> Restarting the pod doesn't guarantee a new public IP, and it's possible customers are using a public gateway (e.g. Cloud NAT) with VM without a public IP address

2. It's also noted that this comes up quite frequently - I think on larger instances running up against the limit is quite common and even expected, and Sourcegraph should for the most part continue working even if it exhausts its rate limits. It becomes a _critical_ issue only if the rate limit is exhausted immediately after a rate limit reset. We can detect this by checking to see if the rate limit is below the threshold for ~most of a window.

## Test plan

n/a

### CODE REVIEW

File: monitoring/definitions/repo_updater.go [hunk 1]
```diff
@@ -392,29 +392,44 @@ func RepoUpdater() *monitoring.Dashboard {
...
 							Query:       `max by (name) (src_github_rate_limit_remaining_v2{resource="graphql"})`,
 							// 5% of initial limit of 5000
-							Critical:  monitoring.Alert().LessOrEqual(250),
-							Panel:     monitoring.Panel().LegendFormat("{{name}}"),
-							Owner:     monitoring.ObservableOwnerRepoManagement,
-							NextSteps: `Try restarting the pod to get a different public IP.`,
+							Warning: monitoring.Alert().LessOrEqual(250),
+							// Critical if most of a 60-minute reset window is spend below
+							// the threshold.
+							Critical: monitoring.Alert().LessOrEqual(250).For(50 * time.Minute),
```
Comment (bobheadxi) [inline monitoring/definitions/repo_updater.go:398]: If the threshold is in critical levels for 50 minutes, that means the rate limit was exhausted within 10 minutes of a rate limit refresh - probably more likely to be a critical issue

**Review by michaellzc — APPROVED** [review]
I remember reading something about admin can turn on a flag to utilize user oauth token to improve permission syncing which reduces the chance the site-level access token is getting rate-limited?

could this be one of the potential next step?

**Review by ryanslade — APPROVED** [review]

**Review by sashaostrikov — APPROVED** [review]

File: monitoring/definitions/repo_updater.go [hunk 2]
```diff
@@ -392,29 +392,44 @@ func RepoUpdater() *monitoring.Dashboard {
...
+							Panel:    monitoring.Panel().LegendFormat("{{name}}"),
+							Owner:    monitoring.ObservableOwnerRepoManagement,
+							NextSteps: `
+								- Consider creating a new token under a dedicated machine user to reduce rate limit pressure.
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply monitoring/definitions/repo_updater.go]: Added a note! 39c8570a02307704ddd6957a90ebad2a0205805d

**Comment by bobheadxi** [comment]
> I remember reading something about admin can turn on a flag to utilize user oauth token to improve permission syncing which reduces the chance the site-level access token is getting rate-limited?
>
> could this be one of the potential next step?

I don't recall anything related to this and a quick search doesn't surface anything, so maybe someone can follow up with a more concrete recommendation 👍

**Pull request merged**